src/conwin.f

```diff
@@ -1,66 +1,51 @@
+C> for an element identified in the user string as a conditional node.
 C>
-C> USAGE:    CALL CONWIN (LUN, INC1, INC2)
-C>   INPUT ARGUMENT LIST:
-C>     LUN      - INTEGER: I/O STREAM INDEX INTO INTERNAL MEMORY ARRAYS
-C>     INC1     - INTEGER: SUBSET BUFFER START INDEX 
-C>     INC2     - INTEGER: SUBSET BUFFER ENDING INDEX
+C> ### Program History Log
+C> Date | Programmer | Comments
+C> -----|------------|---------
+C> 1994-01-06 | J. Woollen | Original author.
+C> 1998-07-08 | J. Woollen | Improved machine portability.
+C> 1999-11-18 | J. Woollen | Increase number of open BUFR FILES.
+C> 2003-11-04 | S. Bender  | Added remarks/bufrlib routine interdependencies.
+C> 2003-11-04 | D. Keyser  | MAXJL increased from 15000 to 16000; unified/portable for WRF.
+C> 2010-04-27 | J. Woollen | correct logical flaw and add documentation.
+C> 2014-12-10 | J. Ator    | use modules instead of common blocks.
 C>
-C>   OUTPUT ARGUMENT LIST:
-C>     INC1     - INTEGER: SUBSET BUFFER START INDEX 
-C>     INC2     - INTEGER: SUBSET BUFFER ENDING INDEX 
+C> @author Woollen @date 1994-01-06
+      
+C> This subroutine searches consecutive subset buffer segments
+C> for an element identified in the user string as a conditional node
+C> (i.e. an element which must meet a condition in order to be read
+C> from or written to a data subset). If a conditional element is
+C> found and it conforms to the condition, then the internal subset
+C> buffer indices of the "window" (see below remarks) are returned to
+C> the caller for processing.
+C> 
+C> The four conditions which can be exercised are:
+C> - '<' - less than
+C> - '>' - greater than
+C> - '=' - equal 
+C> - '!' - not equal
 C>
-C> REMARKS:
+C> Each condition in a string is applied to one element, and all
+C> conditions are 'and'ed to evaluate an outcome. For example, if the
+C> condition string is: "POB<500 TOB>30 TQM<4" then the only levels of
+C> data read or written are those with pressure lt 500 mb, temperature
+C> gt 30 deg, and temperature quality mark < 4.
 C>
-C>    SEE THE DOCBLOCK IN BUFR ARCHIVE LIBRARY SUBROUTINE GETWIN FOR AN
-C>    EXPLANATION OF "WINDOWS" WITHIN THE CONTEXT OF A BUFR DATA SUBSET.
+C> See getwin() for an explanation of "windows" within the context of a
+C> BUFR data subset.
 C>	
-C>    FUNCTION CONWIN WORKS WITH FUNCTION INVCON TO IDENTIFY SUBSET
-C>    BUFFER SEGMENTS WHICH CONFORM TO THE SET OF CONDITIONS.  
-C>
-C>    THIS ROUTINE CALLS:        GETWIN   INVCON
-C>    THIS ROUTINE IS CALLED BY: UFBEVN   UFBIN3   UFBRW
-C>                               Normally not called by any application
-C>                               programs.
+C> Function conwin() works with function invcon() to identify subset
+C> buffer segments which conform to the set of conditions.
+C>	
+C> @param[in] LUN integer I/O stream index into internal memory arrays.
+C> @param[inout] INC1 integer subset buffer start index 
+C> @param[inout] INC2 integer subset buffer ending index
```

Comment:
I may be wrong, but I believe INC1 is actually only ever an output argument and is never an input argument to this routine.  I realize this is how it was listed in the original documentation, but if I'm right it would be nice to correct this as we're going through and updating all of these docblocks anyway.  @jack-woollen knows this subroutine better than I do and can probably tell us for sure.

Either way, I believe it's correct that INC2 is both an input and output argument, but I'll also let Jack confirm that too ;-)

Reply:
@jbathegit, true, INC1 is only an outgoing argument of both conwin and getwin.